httpclient5/src/main/java/org/apache/hc/client5/http/impl/io/PoolingHttpClientConnectionManager.java

```diff
@@ -243,6 +245,16 @@ private InternalConnectionEndpoint cast(final ConnectionEndpoint endpoint) {
         throw new IllegalStateException("Unexpected endpoint class: " + endpoint.getClass());
     }
 
+    private ConnectionConfig resolveConnectionConfig(final HttpRoute route) {
+        final ConnectionConfig connectionConfig = connectionConfigResolver != null ? connectionConfigResolver.resolve(route) : null;
+        return connectionConfig != null ? connectionConfig : ConnectionConfig.DEFAULT;
+    }
+
+    private SocketConfig resolveSocketConfig(final HttpRoute route) {
+        final SocketConfig socketConfig = socketConfigResolver != null ? socketConfigResolver.resolve(route) : null;
+        return socketConfig != null ? socketConfig : SocketConfig.DEFAULT;
```

Comment:
These two methods should use a single volatile read on the resolver as well.

Reply:
@carterkozak Corrected.